build/images/Dockerfile.build.agent.coverage

```diff
@@ -24,6 +24,8 @@ RUN go mod download
 
 COPY . /antrea
 
+RUN chown -R root:root /antrea
```

Comment:
Instead of running a separate chown command after copying files, can prefer to use Docker's COPY --chown option, which allows to set file ownership during the copy process  !
COPY --chown=root:root . /antrea

BTW antrea directory ownership is by default root

Reply:
@luolanzone let me try if I can reproduce locally with the latest Docker

>  In the newer Docker (e.g. 27.3.1), the COPY directive will keep the source ownership info which might be 'jenkins' or other users.

That doesn't sound right. This is what I see in the Dockerfile reference documentation:

> All files and directories copied from the build context are created with a UID and GID of 0 unless the optional --chown flag specifies a given username, groupname, or UID/GID combination to request specific ownership of the copied content.

I also don't see any reference to a change in behavior in the Dockerfile release notes.

Note that such a behavior (preserving owner information) would be very bad IMO: it would completely kill build reproducibility. You are supposed to be able to build the same docker image from everywhere.

Reply:
Not able to reproduce with this docker version:
```
$ docker version                                                                                                              
Client: Docker Engine - Community
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:41:00 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          27.3.1
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.22.7
  Git commit:       41ca978
  Built:            Fri Sep 20 11:41:00 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.22
  GitCommit:        7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c
 runc:
  Version:          1.1.14
  GitCommit:        v1.1.14-0-g2c9f560
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Reply:
Hi @antoninbas let me send you the new testbed info, which you may help to double check on it. If this is not the case, then I believe some other configs changed the behavior.

Reply:
> Instead of running a separate chown command after copying files, can prefer to use Docker's COPY --chown option, which allows to set file ownership during the copy process ! COPY --chown=root:root . /antrea
> 
> BTW antrea directory ownership is by default root

This doesn't work when I tried yesterday `COPY --chown=root:root . /antrea`, so I used RUN to change the ownership. But let's double check before delivering the fix.

Reply:
It turns out the buildx cache caused the difference behaviors. After I run `docker buildx prune`, the build command 'make build-controller-ubuntu' run successfully and the new image is built. Not sure why the cache caused this, but I think we can close this PR after we perform the cache cleanup and verify the kind job on the new testbed. 
@KMAnju-2021 please go ahead. Thanks.